plugins/outputs/microsoft_fabric/microsoft_fabric.go

```diff
@@ -0,0 +1,126 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package microsoft_fabric
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type fabricOutput interface {
+	Init() error
+	Connect() error
+	Write(metrics []telegraf.Metric) error
+	Close() error
+}
+
+type MicrosoftFabric struct {
+	ConnectionString string          `toml:"connection_string"`
+	Log              telegraf.Logger `toml:"-"`
+	Timeout          config.Duration `toml:"timeout"`
+
+	eventhouse   *eventhouse
+	eventstream  *eventstream
+	activePlugin fabricOutput
+}
+
+func (*MicrosoftFabric) SampleConfig() string {
+	return sampleConfig
+}
+
+func (m *MicrosoftFabric) Init() error {
+	connectionString := m.ConnectionString
+
+	if connectionString == "" {
+		return errors.New("endpoint must not be empty")
+	}
+
+	if strings.HasPrefix(connectionString, "Endpoint=sb") {
+		m.Log.Info("Detected EventStream endpoint, using EventStream output plugin")
+		eventstream := &eventstream{}
+		eventstream.connectionString = connectionString
+		eventstream.log = m.Log
+		eventstream.timeout = m.Timeout
+		if err := eventstream.parseconnectionString(connectionString); err != nil {
+			return fmt.Errorf("parsing connection string failed: %w", err)
+		}
+		m.eventstream = eventstream
+		if err := m.eventstream.Init(); err != nil {
+			return fmt.Errorf("initializing EventStream output failed: %w", err)
+		}
+		m.activePlugin = eventstream
+	} else if isKustoEndpoint(strings.ToLower(connectionString)) {
```

Comment:
How about moving this function to a function called `detectEventStream(cs string) bool` in `event_stream.go`?